src/Control/Cookie.php

```diff
@@ -92,4 +100,37 @@ public static function force_expiry($name, $path = null, $domain = null, $secure
     {
         return self::get_inst()->forceExpiry($name, $path, $domain, $secure, $httpOnly);
     }
+
+    /**
+     * Validate if the samesite value for a cookie is valid for the current request.
+     *
+     * Logs a warning if the samesite value is "None" for an insecure request.
+     * @throws LogicException if the value is not "Strict", "Lax", or "None".
+     */
+    public static function validateSameSite(string $sameSite): void
+    {
+        $validValues = [
+            'Strict',
+            'Lax',
+            'None',
+        ];
+        if (!in_array($sameSite, $validValues)) {
+            throw new LogicException('samesite must be "Strict", "Lax", or "None"');
+        }
+        if ($sameSite === 'None' && !Director::is_https(self::getRequest())) {
+            Injector::inst()->get(LoggerInterface::class)->warning('Cookie samesite cannot be "None" for insecure requests.');
+        }
+    }
+
+    /**
+     * Get the current request, if any.
+     */
+    private static function getRequest(): ?HTTPRequest
+    {
+        $request = null;
+        if (Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        return ($request instanceof NullHTTPRequest) ? null : $request;
```

Comment:
A `NullHTTPRequest` will _always_ have its `scheme` set to "http". This means Director won't fall back to checking `default_base_url` because it expects a real request to be passed in with the scheme used for that request. I don't know if that's a bug in `Director` or not.... I don't know what implications that would have for anything else if we did change it there.

Reply:
I've added a comment above this line explaining why we do this.